### ISSUE TITLE
TRAIN-06: standardize encoder freezing

### ISSUE THREAD

**Issue by LorenzoCiarpa** [huggingface/lerobot#4687]
### Ticket Type

💡 Feature Request / Improvement

### Environment & System Info

```Shell

```

### Description

I would like to pick up **TRAIN-06** (*Standardize encoder freezing*) from the 0.7.0
community roadmap (#3832).

**Current state.** 12 policies freeze submodules, through 6 naming conventions
(`freeze_*`, `train_*_only`, `train_*`, `tune_*`, `finetune_*`, `train_mode_vlm`) and 5
mechanisms (inline in `__init__`, `_set_requires_grad()`, `set_requires_grad()`,
`set_trainable_parameters()` + `set_frozen_modules_to_eval_mode()`, and a module-level
`freeze_image_encoder()` helper).

Three things are concretely wrong:

**1. "Frozen" does not imply "in eval mode" — and 5 policies lose it.**
Freezing means both no gradients *and* the module staying in eval mode, so BatchNorm running statistics stop updating and dropout stays off.
Policies that do not override `train()` therefore thaw their "frozen" encoder on every cycle:

| policy | restores eval mode after `policy.train()` |
|---|---|
| pi0, pi05, smolvla, evo1, molmoact2 | yes (each with its own override) |
| groot | yes, via `set_frozen_modules_to_eval_mode()` called from `forward` |
| **xvla, vla_jepa, fastwam, gaussian_actor, multi_task_dit** | **no** |

`gaussian_actor` is the clearest case: it loads a **CNN** encoder (`modeling_gaussian_actor.py:169`, with an explicit `raise` if the architecture is not a CNN), sets `requires_grad = False`, and never puts it back in eval — so the BatchNorm running statistics keep drifting on an encoder the user believes is frozen.

The problem is real and already acknowledged elsewhere in the codebase: ACT avoids it by building its ResNet backbone with `norm_layer=FrozenBatchNorm2d` (`policies/act/modeling_act.py:34,331`). So there are three different in-tree answers to the same question.

**2. `get_optim_params()` has no contract.** Declared `-> dict` in `policies/pretrained.py:216`, but the 21 implementations return a generator (`self.parameters()`), a list of param groups, or a dict — and only 7 of them filter on `requires_grad`. Meanwhile `optim/optimizers.py:188` requires a `named_parameters()` dict and filters itself, so whether freezing reaches the optimizer depends on which optimizer you picked.

**3. pi0 and pi05 are byte-identical here.** `_set_requires_grad()` + `train()` in
`pi0/modeling_pi0.py:270-286` and `pi05/modeling_pi05.py:370-386` diff to nothing.

**Proposed API.** The policy *declares* what is frozen; `PreTrainedPolicy` *enforces* how:

```python
def get_frozen_modules(self) -> dict[str, nn.Module]:   # dotted path -> submodule; default {}
def apply_freezing(self) -> None:                       # requires_grad_(False) + eval()
def set_module_trainable(self, path: str, trainable: bool) -> None:   # the unfreeze side
def train(self, mode: bool = True):                     # super().train(mode), then re-eval frozen
```

Why this shape:
- it makes the frozen set **introspectable**, so the contract can be tested generically
  instead of being an implementation detail of each `__init__`;
- it uses `nn.Module.get_submodule()`, so it adds no new concept;
- it covers the *unfreezing* half that TRAIN-06 asks for;
- it is opt-in: a policy that overrides nothing keeps today's behaviour exactly.

**Proposed PR split**, smallest first, each independently revertable:

- **PR A — foundation, no behaviour change.** The four methods on `PreTrainedPolicy`,
  adopted in pi0 and pi05 (deleting the duplicated blocks), plus a contract test. Related PR: #4688.
- **PR B — the actual fix.** xvla, gaussian_actor, vla_jepa, fastwam, multi_task_dit get
  the eval-mode guarantee they are missing. This one *does* change behaviour, per policy,
  with a test each.
- **PR C — the involved ones.** smolvla (whose "avoid unused params in DDP" layer
  freezing must be preserved), groot (map its existing mechanism onto the API), evo1
  (tri-state flags), molmoact2 (`train_mode_vlm` enum + LoRA).
- **PR D — `get_optim_params` contract.** Only if A–C land and you want it; it is the one
  with real regression risk.

### Context & Reproduction

The eval-mode loss, reduced to its mechanism — a policy with a frozen encoder that
contains a BatchNorm, put through the cycle the training loop actually performs:

```python
policy = ...            # any policy whose config freezes an encoder
policy.train()          # trainer, start of training
policy.eval()           # trainer, evaluation step
policy.train()          # trainer, lerobot_train.py:809 — back to training

assert not policy.encoder.training   # fails today for xvla / vla_jepa / fastwam /
                                     # gaussian_actor / multi_task_dit
```

On a minimal policy built for the test, one forward pass after that cycle moves the
frozen BatchNorm's `running_mean` by 4e-2 — silently, since no gradient is involved.

Equivalence check I used for PR A (pi0 and pi05, 4 flag combinations each): dump
`{parameter: requires_grad}` and `{module: training}` after `policy.train()`, on `main`
and on the branch, and diff. The two 1.3 MB dumps are identical, which is the evidence
that PR A moves the mechanism without moving the behaviour.

### Relevant logs or stack trace

```Shell

```

### Checklist

- [x] I have searched existing tickets to ensure this isn't a duplicate.
- [x] I am using the latest version of the `main` branch.
- [x] I have verified this is not an environment-specific problem.
